src/v/config/configuration.cc

```diff
@@ -2332,6 +2332,16 @@ configuration::configuration()
       // Enough for a >1TiB cache of 16MiB objects.  Decrease this in case
       // of issues with trim performance.
       100000)
+  , cloud_storage_cache_trim_carryover(
+      *this,
+      "cloud_storage_cache_trim_carryover",
+      "The cache performs a recoursive directory walk during the cache trim. "
+      "The information obtained during the walk can be carried over to the "
+      "next trim operation. This parameter sets a limit on number of objects "
+      "that can be carried over from one trim to next. This allows cache to "
+      "quickly unblock readers before starting the directory walk.",
+      {.needs_restart = needs_restart::no, .visibility = visibility::tunable},
+      512)
```

Comment:
Perhaps this value should be larger. On systems where we are reaching obj. count we have upwards of 100k objects. 

Not for this PR but maybe we can optimistically use much more available units from materialized resources and then trim this list to release units on demand if the readers need it, like a ballast file.